python/triton/compiler/compiler.py

```diff
@@ -85,12 +85,17 @@ def optimize_ttgir(mod, num_stages, arch):
         pm.add_tritongpu_accelerate_matmul_pass(80)
     pm.add_tritongpu_remove_layout_conversions_pass()
     pm.add_tritongpu_optimize_dot_operands_pass()
-    pm.add_tritongpu_pipeline_pass(num_stages)
+    if num_stages == 0 and is_hip() and gpu_has_mfma():
```

Comment:
I am limiting this now to an odd enablement. Once it is proven and tested across different workloads (ie. FA also) it may be enabled by default for hip or conditionally with some setting.

Reply:
This is really odd. So to disable both stream pipelining and triton's pipelining, we'll need to set `num_stages=1`.